pkg/typesystem/typesystem_test.go

```diff
@@ -12,6 +12,490 @@ import (
 	"github.com/openfga/openfga/pkg/testutils"
 )
 
+type RelationDetails struct {
+	hasEntrypoints bool
+	hasCycle       bool
+}
+
+func TestHasEntrypoints(t *testing.T) {
+	tests := map[string]struct {
+		model         string
+		inputType     string
+		inputRelation string
+		expectError   string
+		expectDetails *RelationDetails
+	}{
+		`undefined_input_relation`: {
+			model: `
+			model
+				schema 1.1
+			type document
+				relations
+					define viewer: [folder]`,
+			inputType:     "document",
+			inputRelation: "unknown",
+			expectError:   "undefined type definition for 'document#unknown'",
+		},
+		`undefined_relation_in_assignable_type`: {
+			model: `
+			model
+				schema 1.1
+			type document
+				relations
+					define viewer: [document#unknown]`,
+			inputType:     "document",
+			inputRelation: "viewer",
+			expectError:   "undefined type definition for 'document#unknown'",
+		},
+		`undefined_computed_userset`: {
+			model: `
+			model
+				schema 1.1
+			type document
+				relations
+					define viewer: unknown`,
+			inputType:     "document",
+			inputRelation: "viewer",
+			expectError:   "undefined type definition for 'document#unknown'",
+		},
+		`undefined_tupleset`: {
```

Comment:
Please add one more test for undefined computed relation on tupleset target.

```
type user

type folder
  relations
    define owner: [user]

type document
  relations
    define parent: [folder]
     define viewer: viewer from parent
    
```

Reply:
this should throw an error, right? it's currently not 🤔